### PR TITLE
2024 draft

### PR DESCRIPTION
**Pull request by d-w-liu**
Reorganization of folder layout to match new ToC layout; also added placeholders for new files (that have no content).

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -153,7 +153,7 @@ docs
 
 # Jupyterlab
 **/_build
-**.ipynb
+#**.ipynb Actually, pls send this right now
```
Comment (saskiad) [inline .gitignore:156]: I believe there's a reason we don't want ipynb - I'm not ready to undo this until we are sure it's not going to backfire.

File: databook/physiology/np-ultra/np-ultra.md [hunk 2]
```diff
@@ -0,0 +1,5 @@
+# Neuropixels Ultra
+
```
Comment (saskiad) [inline databook/physiology/np-ultra/np-ultra.md:2]: We'll want to adjust this text but as a placeholder this is fine

File: databook/stimuli/stim-passive.md [hunk 3]
```diff
@@ -0,0 +1,3 @@
+# Passive Stimuli
```
Comment (saskiad) [inline databook/stimuli/stim-passive.md:1]: This should be "Passive visual stimuli"

File: databook/_toc.yml [hunk 4]
```diff
@@ -16,74 +16,96 @@ chapters:
...
-  sections:
-  - file: cell-type-lookup-table/ctlut-session-data
-  - file: cell-type-lookup-table/ctlut-optotagging
-  - file: cell-type-lookup-table/ctlut-accessing-data
-  - file: cell-type-lookup-table/ctlut-identifying-tagged-units
-- file: visual-behavior/vb-background
-  sections:
-  - file: visual-behavior/VB-Behavior
-    title: Visual Behavior Task
+  - file: physiology/visual-coding-2p/vc2p-background
```
Comment (saskiad) [inline databook/_toc.yml:25]: we want physioloy/ophys first and then physiology/ophys/visual-coding-2p

File: databook/_toc.yml [hunk 5]
```diff
@@ -16,74 +16,96 @@ chapters:
...
-    - file: visual-behavior/VB-BehaviorSessionData
-      title: Behavior Session Data
-  - file: visual-behavior/VB-Ophys
+    - file: physiology/visual-coding-2p/vc2p-dataset
+    - file: physiology/visual-coding-2p/vc2p-session-data
+    - file: physiology/visual-coding-2p/vc2p-stimuli
+    - file: physiology/visual-coding-2p/vc2p-evoked-response
+    - file: physiology/visual-coding-2p/vc2p-cross-session-data
+    - file: physiology/visual-coding-2p/vc2p-analysis
+  # - file: visual-behavior/vb-background                 # This section needs to go somewhere, but not sure where yet; background stuff on visual behavior.
```
Comment (saskiad) [inline databook/_toc.yml:33]: this goes in the stimulus section

File: databook/_toc.yml [hunk 6]
```diff
@@ -16,74 +16,96 @@ chapters:
...
       title: Tutorial Analyzing LFP data
-- file: microns-em/em-background
+  - file: physiology/cell-type-lookup-table/ctlut-background
+    sections:
+    - file: physiology/cell-type-lookup-table/ctlut-session-data
+    - file: physiology/cell-type-lookup-table/ctlut-optotagging
+    - file: physiology/cell-type-lookup-table/ctlut-accessing-data
+    - file: physiology/cell-type-lookup-table/ctlut-identifying-tagged-units
+  # New files here
+  - file: physiology/np-ultra/np-ultra                     # Also a placeholder that needs actual content
```
Comment (saskiad) [inline databook/_toc.yml:85]: physiology/ephys/np-ultra
then physiology/ephys/np-ultra/np-ultra-background

File: databook/_toc.yml [hunk 7]
```diff
@@ -16,74 +16,96 @@ chapters:
...
-  - file: microns-em/em-caveclient-setup.md
-  - file: microns-em/em-querying-tables.md
-  - file: microns-em/key-tables.md
-  - file: microns-em/em-volume-data.md
-  - file: microns-em/em-skeletons.md
-  - file: microns-em/em-meshes.md
-  - file: microns-em/em-functional_data.md
-  - file: microns-em/em-coordinates.md
-  - file: microns-em/em-ultrastructure.md
+    - file: stimuli/stim-passive
```
Comment (saskiad) [inline databook/_toc.yml:88]: see name changes below

File: databook/stimuli/stim-behavior.md [hunk 8]
```diff
@@ -0,0 +1,3 @@
+# Behavior Stimuli
```
[the remark this answers is not in the document]
Comment (saskiad) [reply databook/stimuli/stim-behavior.md:1]: and this is where the Visual Behavior task information should be move to

File: databook/_toc.yml [hunk 9]
```diff
@@ -16,74 +16,96 @@ chapters:
...
+    - file: stimuli/stim-behavior
```
Comment (saskiad) [inline databook/_toc.yml:89]: ditto

File: databook/physiology/mFISH/mFISH-Dataset.md [hunk 10]
```diff
@@ -0,0 +1,3 @@
+# mFish-Dataset
```
Comment (saskiad) [inline databook/physiology/mFISH/mFISH-Dataset.md:1]: see name changes above.

File: databook/physiology/mFISH/mFISH.md [hunk 11]
```diff
@@ -0,0 +1,5 @@
+# mFISH
```
Comment (saskiad) [inline databook/physiology/mFISH/mFISH.md:1]: again, see changes in toc above.

File: databook/_toc.yml (hunk 9 above)
Comment (saskiad) [reply databook/_toc.yml:89]: also add a stimuli/optotagging page

**Review by saskiad — CHANGES_REQUESTED** [review]
See comments - happy to discuss.
Can we remove the computational tutorials from the PR and make them a separate PR. It's too hard to review multiple issues simultaneously.
I'm not sold on allowing ipynb because there was an active reason we chose to exclude them, so we need a different way of doing that.

File: .gitignore (hunk 1 above)
Comment (d-w-liu) [reply .gitignore]: Reverted .gitignore to exclude ipynb again

File: databook/stimuli/stim-behavior.md (hunk 8 above)
[the remark this answers is not in the document]
Comment (d-w-liu) [reply databook/stimuli/stim-behavior.md]: Renamed file and header

File: databook/stimuli/stim-passive.md (hunk 3 above)
Comment (d-w-liu) [reply databook/stimuli/stim-passive.md]: Renamed file and changed header

File: databook/_toc.yml (hunk 4 above)
Comment (d-w-liu) [reply databook/_toc.yml]: Adjusted file structure and updated the table of contents

File: databook/_toc.yml (hunk 5 above)
Comment (d-w-liu) [reply databook/_toc.yml]: Adjusted the file structure and updated ToC

File: databook/_toc.yml (hunk 6 above)
Comment (d-w-liu) [reply databook/_toc.yml]: Adjusted file structure and updated ToC

File: databook/_toc.yml (hunk 7 above)
Comment (d-w-liu) [reply databook/_toc.yml]: Updated names

File: databook/_toc.yml [hunk 12]
```diff
@@ -16,74 +16,96 @@ chapters:
...
       title: Ophys Experiment Data
-    - file: visual-behavior/VBO-Tutorial-Compare_trial_types
+    - file: physiology/visual-behavior/VBO-Tutorial-Compare_trial_types
       title: Tutorial Comparing Trial Types
-    - file: visual-behavior/VBO-Tutorial-Downloading_data_and_exploring_the_manifest
+    - file: physiology/visual-behavior/VBO-Tutorial-Downloading_data_and_exploring_the_manifest
       title: Tutorial Downloading Data
-  - file: visual-behavior/VB-Neuropixels
+  # New file here
+  - file: physiology/mFISH/mFISH                                     # No idea what this is, but here we are. May get restructured again
```
[the remark this answers is not in the document]
Comment (d-w-liu) [reply databook/_toc.yml]: Updated names and adjusted ToC

File: databook/physiology/mFISH/mFISH-Dataset.md (hunk 10 above)
Comment (d-w-liu) [reply databook/physiology/mFISH/mFISH-Dataset.md]: Updated name and adjusted ToC

File: databook/physiology/mFISH/mFISH.md (hunk 11 above)
Comment (d-w-liu) [reply databook/physiology/mFISH/mFISH.md]: Updated name and adjusted ToC

File: databook/physiology/np-ultra/np-ultra.md (hunk 2 above)
Comment (d-w-liu) [reply databook/physiology/np-ultra/np-ultra.md]: Left as is as placeholder

File: databook/_toc.yml [hunk 13]
```diff
@@ -16,74 +16,95 @@ chapters:
...
+  - file: stimuli/passive-visual-stimuli
+  - file: stimuli/change-detection-task
+  - file: stimuli/visual-behavior/vb-background                 # This section needs to go somewhere, but not sure where yet; background stuff on visual behavior.
+    sections:
+    - file: stimuli/visual-behavior/VB-Behavior
+      title: Visual Behavior Task
+      sections:
+      - file: stimuli/visual-behavior/VB-BehaviorSessionData
+        title: Behavior Session Data
+    - file: stimuli/visual-behavior/VB-Ophys
```
Comment (saskiad) [inline databook/_toc.yml:67]: VB-Ophys, VBO-Dataset, VBO-ExperimentData, VBO_Tutorial... (X2)
VB-Neuropixels, VBN-etc
these all need to be in the physiology section, not the stimuli section. The only one that should be in the stimulus section is VB-Behavior.

**Review by saskiad — CHANGES_REQUESTED** [review]
Most of the Visual behavior content needs to be in the physiology section - separated into ophys and ephys as indicated. Only information about the behavioral task should be in the stimulus section.

File: databook/_toc.yml [hunk 14]
```diff
@@ -11,79 +11,95 @@ chapters:
...
+      - file: physiology/ephys/visual-coding/vcnp-receptive-fields
+      - file: physiology/ephys/visual-coding/vcnp-lfp
+      - file: physiology/ephys/visual-coding/vcnp-optotagging
+      - file: physiology/ephys/visual-coding/vcnp-session
+      - file: physiology/ephys/visual-coding/vcnp-cheatsheet
+      - file: physiology/ephys/visual-coding/vcnp-links
+    - file: physiology/ephys/visual-behavior/VB-Neuropixels
+      title: Visual Behavior Neuropixels
+      sections:
+      - file: physiology/visual-behavior/VBN-Dataset
```
Comment (saskiad) [inline databook/_toc.yml:60]: all these VBN pages need to be in the ephys sub-folder:
physiology/ephys/visual-behavior/VBN-xxx

File: databook/_toc.yml [hunk 15]
```diff
@@ -11,79 +11,95 @@ chapters:
...
+    - file: anatomy/microns-em/em-caveclient-setup.md
+    - file: anatomy/microns-em/em-querying-tables.md
+    - file: anatomy/microns-em/key-tables.md
+    - file: anatomy/microns-em/em-volume-data.md
+    - file: anatomy/microns-em/em-skeletons.md
+    - file: anatomy/microns-em/em-meshes.md
+    - file: anatomy/microns-em/em-functional_data.md
+    - file: anatomy/microns-em/em-coordinates.md
+    - file: anatomy/microns-em/em-ultrastructure.md
+  #- file: This will be something on Single Cell Whole Brain Anatomy once we've determined a name
```
Comment (saskiad) [inline databook/_toc.yml:103]: let's add the anatomy/single-cell-morphology section and it'll be a placeholder as we add that content

File: databook/physiology/stimuli/passive-visual-stimuli/visual-stimuli-list.md [hunk 16]
```diff
@@ -0,0 +1,48 @@
+# Visual Stimuli
+
+# Probably want figures of the stimuli
+
```
Comment (saskiad) [inline databook/physiology/stimuli/passive-visual-stimuli/visual-stimuli-list.md:4]: Hold off on this - if the idea is not to use the vc2p stimulus page this needs to be much more generic. It's fine to leave this - we can edit it later.

File: databook/physiology/stimuli/passive-visual-stimuli/visual-stimuli-np.md [hunk 17]
```diff
@@ -0,0 +1,26 @@
+# Visual Stimuli
```
Comment (saskiad) [inline databook/physiology/stimuli/passive-visual-stimuli/visual-stimuli-np.md:1]: No, we're only having one page of passive stimuli in the stimulus section.

File: databook/_toc.yml [hunk 18]
```diff
@@ -11,79 +11,95 @@ chapters:
...
-    - file: visual-behavior/VBN-Tutorial-Aligning_Neural_Data_to_Stimuli.ipynb
-      title: Tutorial Aligning Neural Data to Stimuli
-    - file: visual-behavior/VBN-Tutorial-Aligning_Behavior_Data_to_Task_Events.ipynb
-      title: Tutorial Aligning Behavioral Data to Task Events
-    - file: visual-behavior/VBN-Tutorial-Analyzing_LFP_Data.ipynb
-      title: Tutorial Analyzing LFP data
-- file: microns-em/em-background
+    - file: physiology/stimuli/passive-visual-stimuli/visual-stimuli-list                    # Also a placeholder that needs actual content
+      sections:
+      - physiology/stimuli/passive-visual-stimuli/visual-stimuli-np
```
Comment (saskiad) [inline databook/_toc.yml:79]: no, there is one page for passive visual stimuli and one page for behavior task.

**Review by saskiad — CHANGES_REQUESTED** [review]
getting closer - still a few things to fix

**Review by saskiad — APPROVED** [review]

**Pull request merged**